sdks/typescript/src/index.ts

```diff
@@ -0,0 +1,83 @@
+import axios, { AxiosError } from "axios";
+import {
+  IEtherealPulse,
+  ISendEmailRequest,
+} from "./interfaces/email-services.interface";
+
+class EtherealPulse implements IEtherealPulse {
+  private apiKey: string;
+  private baseUrl: string;
+
+  constructor(apiKey: string) {
+    this.apiKey = apiKey;
+    this.baseUrl = process.env.BASE_URL || "https://api.yourdomain.com/v1";
```

Comment:
I don't think this needs to be in the constructor and maybe could be defined in a variable